runtime/etc/Makefile.mli-common

```diff
@@ -0,0 +1,65 @@
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Makefile.mli-common
+#
+
+include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.include
+
+# $(call getHostBaseValue,value)
+define getHostBaseValue
+$(shell CHPL_MAKE_HOME=$(CHPL_MAKE_HOME) \
+  CHPL_MAKE_HOST_TARGET=--host \
+  make -f $(CHPL_MAKE_HOME)/make/Makefile.base \
+  printQ-$1)
+endef
+
+# $(call runHostRule,file,rule)
+define runHostRule
+$(shell CHPL_MAKE_HOME=$(CHPL_MAKE_HOME) \
+  CHPL_MAKE_HOST_TARGET=--host \
+  make -f $(CHPL_MAKE_HOME)/$1 \
+  $2)
+endef
+
+#
+# In multilocale library compilation, the client library is compiled with the
+# host compiler (CHPL_HOST_COMPILER), while the Chapel server (what is
+# launched on the compute nodes) is compiled with the target compiler
+# (CHPL_TARGET_COMPILER).
+#
+HOST_CC=$(call getHostBaseValue,CC)
+TARG_CC=$(CC)
+
+#
+# These two sets of flags are used when compiling the launcher object files
+# in `Makefile.launcher`. We mimick them here when compiling the client
+# library.
+#
+HOST_CFLAGS=$(call runHostRule,runtime/etc/Makefile.include,printcflags)
```

Comment:
Personally, I find trying to stuff host and target flags into the same file confusing. Can we just split out the mli makefile into host and target files? 

That way we could just continue to use the normal `CHPL_MAKE_HOST_TARGET = --{host,target}` scheme instead of trying to get both host and target values in one makefile

Reply:
> Can we just split out the mli makefile into host and target files?

I think I agree with this.  The coding to handle both host and target in one place is a bit hard to follow, for me.  It appears that if we split into, say:

    Makefile.mli-host-shared
    Makefile.mli-host-static
    Makefile.mli-target-shared
    Makefile.mli-target-static

that we could both get rid of Makefile.mli-common and simplify the coding in the remaining 4 files by using `$(CC)` in place of `$(HOST_CC)` and such.